targets/TARGET_NORDIC/TARGET_NRF5x/serial_api.c

```diff
@@ -1697,7 +1697,7 @@ void serial_rx_asynch(serial_t *obj, void *rx, size_t rx_length, uint8_t rx_widt
     MBED_ASSERT(obj);
     MBED_ASSERT(rx_width == 8);
     MBED_ASSERT(rx_length < 256);
-    MBED_ASSERT(char_match == SERIAL_RESERVED_CHAR_MATCH);
+    MBED_ASSERT(char_match != SERIAL_RESERVED_CHAR_MATCH);
```

Comment:
I am not certain why it asserts even on the condition as it is ? As default value for char_match is this reserved character, thus if read without character match is used, then char_match contains reserved value (=not used) ?

Reply:
Ah good point, perhaps this assert needs to be just removed entirely... I haven't checked if anything similar exists in the previous nordic platforms.

Reply:
I think so. I understand the rest of the parameters there to be checked (not supported length , width etc) but asserting on char match does not make sense to me. Please verify

Reply:
The assert is there to check that the default value (SERIAL_RESERVED_CHAR_MATCH) hasn't been changed.